### PR TITLE
fix(fixed the majority of todos): fixed todos and added some type def…

### PR DESCRIPTION
**Pull request by Danishkhurshid**
…initions

### CODE REVIEW

File: types/wordpress/menus.d.ts [hunk 1]
```diff
@@ -4,6 +4,6 @@ export interface MenuItemProps {
   parentId?: string;
   label: string;
   path: string;
-  target?: string;
+  target?: "_self" | "_blank" | undefined;
```
Comment (bmarshall511) [inline types/wordpress/menus.d.ts]: @Danishkhurshid This shouldn't be limited, targets can be any kind of string.

File: types/wordpress/blocks.d.ts [hunk 2]
```diff
@@ -65,7 +67,7 @@ export interface GutenbergAntDesignButtonBlockProps extends CoreBlockProps {
     disabled?: boolean;
     ghost?: boolean;
     href?: string;
-    htmlType?: string;
+    htmlType?: "submit"| "button" | "reset";
```
Comment (bmarshall511) [inline types/wordpress/blocks.d.ts]: @Danishkhurshid This shouldn't be limited, can potentially be one of many different types that HTML provides for different elements

File: functions/wordpress/postTypes/getPostTypeStaticPaths.ts [hunk 3]
```diff
@@ -43,7 +44,7 @@ export default async function getPostTypeStaticPaths(postType) {
       };
     })
     // Filter out certain posts with custom routes (e.g., homepage).
-    .filter((post) =>
+    .filter((post: any) =>
```
Comment (bmarshall511) [inline functions/wordpress/postTypes/getPostTypeStaticPaths.ts:48]: @Danishkhurshid Please add a TODO above this line

File: functions/wordpress/postTypes/getPostTypeStaticPaths.ts [hunk 4]
```diff
@@ -31,10 +31,11 @@ export default async function getPostTypeStaticPaths(postType) {
         }
       }
     }
-  }`).then((response) => response?.[pluralName]?.edges ?? []);
+  }`, '').then((response) => response?.[pluralName]?.edges ?? []);
 
+  //@TODO: Assign type for posts if required
   const paths = posts
-    .map((post) => {
+    .map((post: any) => {
```
Comment (bmarshall511) [inline functions/wordpress/postTypes/getPostTypeStaticPaths.ts:38]: @Danishkhurshid Please move the @TODO comment above to the line above here. Ideally, there should be a @TODO for each instance of an issue for better tracking

File: app/components/atomic-design/molecules/Navigation/Navigation.tsx [hunk 5]
```diff
@@ -87,9 +83,9 @@ export interface NavigationItemProps {
   /** Item path. */
   path: string;
   /** Link target. */
-  target?: string;
+  target?: "_blank" | "_self" | undefined;
```
Comment (bmarshall511) [inline app/components/atomic-design/molecules/Navigation/Navigation.tsx]: @Danishkhurshid This shouldn't be limited, can be any string

File: app/components/atomic-design/molecules/Navigation/Navigation.tsx [hunk 6]
```diff
@@ -61,7 +57,7 @@ function NavigationMenu({
               {item.label}
             </Link>
 
-            {item?.children?.items?.length > 0 && (
+            {item?.children?.items && (
```
Comment (bmarshall511) [inline app/components/atomic-design/molecules/Navigation/Navigation.tsx]: @Danishkhurshid If any empty array was provided, wouldn't this still get triggered (which we don't want to happen)?

**Review by bmarshall511 — CHANGES_REQUESTED** [review]
@Danishkhurshid Looks great! Just a handful of things & then we can get it merged.

File: types/wordpress/blocks.d.ts [hunk 7]
```diff
@@ -67,7 +67,7 @@ export interface GutenbergAntDesignButtonBlockProps extends CoreBlockProps {
     disabled?: boolean;
     ghost?: boolean;
     href?: string;
-    htmlType?: "submit"| "button" | "reset";
+    htmlType?: ButtonProps.htmlType;
```
Comment (bmarshall511) [inline types/wordpress/blocks.d.ts]: @Danishkhurshid Needs to be a string as mentioned before, shouldn't be Ant specific

**Comment by bmarshall511** [comment]
@Danishkhurshid Please address the items we spoke about:

1. Look into import the Gutenberg block props from Gutenberg into our blocks definition vs. adding it in individual files
2. Get your local setup & test thoroughly. Using the nextstarter, once you have WP installed and setup, you should be able to go to /wordpress to see the homepage.
3. Finish the work you're currently doing for the component styles
4. Look into how to reuse the scoped component styles vs. adding it to every defined component

File: app/wordpress/[slug]/page.tsx [hunk 8]
```diff
@@ -12,6 +12,9 @@ import formatBlockData from "@/functions/gutenberg/formatBlockData";
 // Import component dependencies
 import Blocks from "@/components/gutenberg/Blocks/Blocks";
 
+// Import TypeScript definitions
+import { GutenbergBlockProps } from "@/types/gutenberg";
+
```
Comment (bmarshall511) [inline app/wordpress/[slug]/page.tsx:17]: @Danishkhurshid Did you look into moving this under the block types we've defined as discussed?

**Review by bmarshall511 — CHANGES_REQUESTED** [review]

File: app/wordpress/[slug]/page.tsx (hunk 8 above)
Comment (Danishkhurshid) [reply app/wordpress/[slug]/page.tsx:17]: @bmarshall511 yes. Need to implicitly add a type here.

File: types/wordpress/blocks.d.ts (hunk 7 above)
Comment (Danishkhurshid) [reply types/wordpress/blocks.d.ts]: @bmarshall511 this file no longer exists.

**Pull request merged**